### ISSUE TITLE
Why are symbols of elf files ignored if there size is equals zero?

### ISSUE THREAD

**Issue by nbosb** [janweinstock/vcml#33]
Is there a reason symbols of size 0 are ignored in `elf_reader`?
https://github.com/janweinstock/vcml/blob/103db7e51d5b7d57e6c86a7064d3d55d95d10043/src/vcml/debugging/elf_reader.cpp#L122-L123
I tried to get a symbol from an elf file that has been skipped in `elf_reader` due to unknown symbol size (`syms[i].st_size == 0`). I think, I my case the size of the symbol was `0` because it was a label in an assembly file (where the size of the "function" was not know).

**Comment by janweinstock** [comment]
Hi Nils,

in the past, we only cared for functions `STT_FUNC` and variables `STT_OBJECT`, both types should have a non-zero size. Therefore we are filtering everything else out. Also, it might lead to trouble if someone needs to calculate function boundaries and finds a zero-size function. Therefore I decided to throw those symbols out. Normally, people should annotate the size of a function in assembler code via `.size`, but some forget to, and I have seen this annotation missing even in kernel code.

I have no strong opinion on this though, feel free to remove this check if it helps you.

Gruss
Jan
